TransferMethod/Sources/View/Widget/TextWidget.swift

```diff
@@ -69,6 +89,183 @@ class TextWidget: AbstractWidget {
+            var finalText = ""
+            var isEscapedCharacter = false
+            var patternIndex = pattern.startIndex
+
+            if let currentText = currentText, !currentText.isEmpty {
+                var patternCharactersToBeWritten = ""
+
+                while true {
+                    let patternRange = patternIndex ..< pattern.index(after: patternIndex)
+                    let currentPatternCharacter = [Character](pattern[patternRange])
+                    let currentTextRange = currentTextIndex ..< currentText.index(after: currentTextIndex)
+                    let currentTextCharacter = String(currentText[currentTextRange])
+
+                    if isEscapedCharacter {
+                        isEscapedCharacter = false
+                        finalText += currentPatternCharacter
+                        patternIndex = pattern.index(after: patternIndex)
+                        if patternIndex >= pattern.endIndex || currentTextIndex >= currentText.endIndex {
+                            break
+                        }
+                        continue
+                    }
+
+                    applyFormatForPatternCharacter(currentPatternCharacter: currentPatternCharacter,
+                                                   currentText: currentText,
+                                                   currentTextCharacter: currentTextCharacter,
+                                                   currentTextIndex: &currentTextIndex,
+                                                   finalText: &finalText,
+                                                   isEscapedCharacter: &isEscapedCharacter,
+                                                   pattern: pattern,
+                                                   patternIndex: &patternIndex,
+                                                   patternCharactersToBeWritten: &patternCharactersToBeWritten)
+
+                    if patternIndex >= pattern.endIndex || currentTextIndex >= currentText.endIndex {
+                        break
+                    }
+                }
+            }
+            return finalText
+        }
+        return inputText
+    }
+
+    // swiftlint:disable function_parameter_count
+    private func applyFormatForPatternCharacter(currentPatternCharacter: [Character],
+                                                currentText: String,
+                                                currentTextCharacter: String,
+                                                currentTextIndex: inout String.Index,
+                                                finalText: inout String,
+                                                isEscapedCharacter: inout Bool,
+                                                pattern: String,
+                                                patternIndex: inout String.Index,
+                                                patternCharactersToBeWritten: inout String) {
```

Comment:
Too many paramteres

Reply:
Yes unfortunately the logic in the method needs them all. I had to refactor this code out as the parent method had become too long and taking this method out meant passing all the parameters needed.